instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/TracingApiCallback.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.kubernetesclient;
+
+import static io.opentelemetry.javaagent.instrumentation.kubernetesclient.KubernetesClientTracer.tracer;
+
+import io.kubernetes.client.openapi.ApiCallback;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.ApiResponse;
+import io.opentelemetry.context.Context;
+import java.util.List;
+import java.util.Map;
+
+public class TracingApiCallback<T> implements ApiCallback<T> {
+  private final ApiCallback<T> delegate;
+  private final Context context;
+
+  public TracingApiCallback(ApiCallback<T> delegate, Context context) {
```

Comment:
should we pass in `parentContext` also for running the callbacks?

Reply:
Good point, done.